### ISSUE TITLE
Support for \lefteqn

### ISSUE THREAD

**Issue by anandkumar89** [KaTeX/KaTeX#3772]
> Before requesting a feature, please search for [existing issues](https://github.com/KaTeX/KaTeX/issues).

**Is your feature request related to a problem? Please describe.**
I want to render 

![](https://i.stack.imgur.com/ZqU4Q.png)

Katex supports overbrace and underprice; but there seems no way to achieve above. Support for \lefteqn will allow this as described below

**Describe the solution you'd like:**
A solution is to use phantom along with lefteqn (described [here](https://tex.stackexchange.com/questions/297/how-can-i-get-an-underbrace-and-an-overbrace-to-partially-overlap-in-an-equation) and below)

\[
0+\lefteqn{\overbrace{\phantom{1+2+3}}}1+
\underbrace{2+3+\overbrace{x+y+z}+
\lefteqn{\overbrace{\phantom{4+5}}}4}+5
\]

**Link to or name of a (La)TeX package that provides the same feature:**
Usually \lefteqn{} is used to break long line of equation and is supported natively. Other packages which achieve this is `outraces`  and `underoverlap` (see [same as above link](https://tex.stackexchange.com/questions/297/how-can-i-get-an-underbrace-and-an-overbrace-to-partially-overlap-in-an-equation))

**Describe alternatives you've considered:**
Currently I don't see any way to achieve same through latex.

**Comment by edemaine** [comment]
I agree that this would be nice, but I fear it's difficult to achieve. @ronkok can confirm, but I don't think this is possible given how we currently implement braces, which autosize to the content being braced via HTML.

**Comment by edemaine** [comment]
But sorry, I misread the original request, and wasn't familiar with `\lefteqn`. As far as I can tell, `\lefteqn{...}` is equivalent to `\mathrlap{\displaystyle ...}`.  So the following works well:

```latex
0+\mathrlap{\overbrace{\phantom{1+2+3}}}1+
  \underbrace{2+3+\overbrace{x+y+z}+
  \mathrlap{\overbrace{\phantom{4+5}}}4}+5
```

![image](https://user-images.githubusercontent.com/2218736/210593994-1e42e098-9058-434b-a280-c3255a907e5f.png)

If this is correct, perhaps we should provide `\lefteqn` as a macro.

**Comment by anandkumar89** [comment]
Thanks for quick response. This solves my purpose. Likely `lefteqn` could be added as macro. for reference, I'd attach some standard behaviour.

## How it is used 
`\lefteqn{eqn}` will have zero width. This allows splitting long equations in eqnarray/align environments. e.g. 

```{latex}
\begin{eqnarray*}
 \lefteqn{w+x+y+z = }\\
   && a+b+c+d+e+\\
   && {}+f+g+h+i
\end{eqnarray*}
```
results in (see [here](http://www.personal.ceu.hu/tex/cookbook.html))

![](http://oziris.ceu.hu/latex/ea80235995b019e612ef52d3630a6b54.png)

[(see this, bottom section)](http://latexref.xyz/eqnarray.html) : The command `\lefteqn` is used for splitting long formulas across lines. It typesets its argument in display style flush left in a box of zero width.

**Comment by ronkok** [comment]
A way to achieve the desired rendering in current KaTeX is via:

```
x = \mathrlap{\underbrace{\phantom{A+B+C}}}A+\mathrlap{\overbrace{\phantom{B+C+D}}}B+C+\underbrace{D+E}
```

... which can be seen at: [link](https://katex.org/?data=%7B%22displayMode%22%3Atrue%2C%22leqno%22%3Afalse%2C%22fleqn%22%3Afalse%2C%22throwOnError%22%3Atrue%2C%22errorColor%22%3A%22%23cc0000%22%2C%22strict%22%3A%22warn%22%2C%22output%22%3A%22htmlAndMathml%22%2C%22trust%22%3Afalse%2C%22macros%22%3A%7B%22%5C%5Cf%22%3A%22%231f(%232)%22%7D%2C%22code%22%3A%22x%20%3D%20%5C%5Cmathrlap%7B%5C%5Cunderbrace%7B%5C%5Cphantom%7BA%2BB%2BC%7D%7D%7DA%2B%5C%5Cmathrlap%7B%5C%5Coverbrace%7B%5C%5Cphantom%7BB%2BC%2BD%7D%7D%7DB%2BC%2B%5C%5Cunderbrace%7BD%2BE%7D%22%7D)

`\lefteqn` probably would still be useful. But not absolutely necessary for this case.

**Comment by ronkok** [comment]
Ah, I see that I have just repeated part of the conversation. Well, never mind.
